runtime/appruntime/apisdk/api/auth_remote.go

```diff
@@ -0,0 +1,163 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/rs/zerolog"
+
+	"encore.dev/appruntime/apisdk/api/transport"
+	"encore.dev/appruntime/exported/config"
+	"encore.dev/appruntime/exported/model"
+	"encore.dev/beta/errs"
+)
+
+// remoteAuthHandler is an AuthHandler that calls a remotely hosted auth handler.
+//
+// This is used when the auth handler is not being hosted in this container, but in
+// another container somewhere else in the cluster.
+type remoteAuthHandler struct {
+	server         *Server        // The server we're running in
+	hostingService config.Service // The service name of the remote auth handler
+	authURL        string         // The URL of the remote auth handler
+	original       AuthHandler
+	logger         zerolog.Logger
+	traceLogs      bool
+}
+
+var _ AuthHandler = (*remoteAuthHandler)(nil)
+
+func (r *remoteAuthHandler) Authenticate(c IncomingContext) (model.AuthInfo, error) {
+	// Quickly check if the auth data is parsable here, if it isn't we can return early
+	// without making a remote call.
+	if err := r.original.ParseAuthData(c); err != nil {
+		return model.AuthInfo{}, err
+	}
+
+	if r.traceLogs {
+		r.logger.Trace().Msg("calling auth handler")
+	}
+
+	// Create the auth request
+	authReq, err := http.NewRequestWithContext(c.ctx, http.MethodPost, r.authURL, nil)
+	if err != nil {
+		r.logger.Err(err).Msg("unable to create auth request")
+		return model.AuthInfo{}, errs.Wrap(err, "unable to create auth request")
+	}
+
+	// Copy over any data which might be needed by the auth handler (query string, headers and cookies)
+	authReq.URL.RawQuery = c.req.URL.RawQuery
+	for k, v := range c.req.Header {
+		// Don't copy the platform auth key across
+		if k == "X-Encore-Auth" {
+			continue
+		}
+
+		for _, vv := range v {
+			authReq.Header.Add(k, vv)
+		}
+	}
+
+	// Add call meta data to the request, so the receiving service can use to allow access to the auth handler
+	// This also passes a shared TraceID to the auth handler, so it and the microservice called after the gateway
```

Comment:
Nit: we don't typically specify "microservice" but just "service"
```suggestion
	// This also passes a shared TraceID to the auth handler, so it and the service called after the gateway
```